From a43ab9f2fed8f56ac0db8d6a4655c05bc0988afe Mon Sep 17 00:00:00 2001
From: Maxim Levitsky <maximlevitsky@xxxxxxxxx>
Date: Tue, 1 Jun 2010 15:43:21 +0300
Subject: [PATCH 90/91] wireless: allow to retrieve the channel set on monitor
 interface This will allow to preserve compatibility
 with userspace Signed-off-by: Maxim Levitsky
 <maximlevitsky@xxxxxxxxx>

---
 net/wireless/chan.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/net/wireless/chan.c b/net/wireless/chan.c
index 17cd0c0..1ba03ba 100644
--- a/net/wireless/chan.c
+++ b/net/wireless/chan.c
@@ -82,9 +82,12 @@ int cfg80211_set_freq(struct cfg80211_registered_device *rdev,
 {
 	struct ieee80211_channel *chan;
 	int result;
+	struct wireless_dev *mon_dev = NULL;
 
-	if (wdev && wdev->iftype == NL80211_IFTYPE_MONITOR)
+	if (wdev && wdev->iftype == NL80211_IFTYPE_MONITOR) {
+		mon_dev = wdev;
 		wdev = NULL;
+	}
 
 	if (wdev) {
 		ASSERT_WDEV_LOCK(wdev);
@@ -131,5 +134,8 @@ int cfg80211_set_freq(struct cfg80211_registered_device *rdev,
 	if (wdev)
 		wdev->channel = chan;
 
+	if (mon_dev)
+		mon_dev->channel = chan;
+
 	return 0;
 }
-- 
1.7.5.4

